hasher-matcher-actioner/hmalib/lambdas/api/actions_api.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import bottle
+import typing as t
+from dataclasses import dataclass, asdict
+from .middleware import jsoninator, JSONifiable, DictParseable
+from hmalib.common.config import HMAConfig, update_config
+from hmalib.common import config as hmaconfig
+from hmalib.common.actioner_models import ActionPerformer
+
+
+@dataclass
+class Action(JSONifiable, DictParseable):
+    name: str
+    config_subtype: str
+    url: str
+    headers: str
+
+    def to_json(self) -> t.Dict:
+        return asdict(self)
+
+    @classmethod
+    def from_dict(cls, d: dict) -> "Action":
+        return cls(
+            d["name"],
+            d["config_subtype"],
+            d["url"] if d["url"] else "",
+            d["headers"] if d["headers"] else "",
+        )
+
+
+@dataclass
+class ActionsResponse(JSONifiable):
+    actions_response: t.List[Action]
+
+    def to_json(self) -> t.Dict:
+        return {
+            "actions_response": [action.to_json() for action in self.actions_response]
+        }
+
+
+@dataclass
+class DeleteActionResponse(JSONifiable):
+    response: str
+
+    def to_json(self) -> t.Dict:
+        return asdict(self)
+
+
+@dataclass
+class CreateActionResponse(JSONifiable):
+    response: str
+
+    def to_json(self) -> t.Dict:
+        return asdict(self)
+
+
+def get_actions_api(hma_config_table: str) -> bottle.Bottle:
+    # The documentation below expects prefix to be '/actions/'
+    actions_api = bottle.Bottle()
+    HMAConfig.initialize(hma_config_table)
+
+    @actions_api.get("/", apply=[jsoninator])
+    def get_actions() -> ActionsResponse:
+        """
+        Returns all action configs.
+        """
+        action_configs = ActionPerformer.get_all()
+        return ActionsResponse(
+            actions_response=[
+                Action.from_dict(config.__dict__) for config in action_configs
+            ]
```

Comment:
Maybe it makes sense to combine `ActionPerformer` and `Action` classes?

Reply:
yes, when I think more about this, looks ```Action``` is not needed